utils/src/foreign_field.rs

```diff
@@ -413,33 +449,55 @@ mod tests {
         let big = secp256k1_modulus();
```

Comment:
A bit unrelated to this PR, but still good to have a for a quick follow-up: generalizing the foreign field, and take a random prime field.